app/controllers/catalog_controller.rb

```diff
@@ -132,9 +132,17 @@ def set_parent_document
       query = "member_ids_ssim:id-#{params['id']}"
       _, result = search_results(q: query, rows: 1)
       @parent_document = result.first if result.first
+    elsif params[:parent_id].nil? && @document.decorated_resource.is_a?(CoinDecorator)
```

Comment:
This seems like a good feature to do for everything. Maybe change it to `@document.decorated_resource.try(:decorated_parent)`